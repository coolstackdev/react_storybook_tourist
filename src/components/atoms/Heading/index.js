import React from 'react'
import T from 'prop-types'
import styled, { css } from 'styled-components'
import { font } from 'styled-theme'
import { palette } from 'styled-tools'

const fontSize = ({ level }) => `${0.75 + (1 * (1 / level))}rem`

const styles = css`
  font-family: ${font('primary')};
  font-weight: ${props => !!props.bold ? 700 : 500};
  font-size: ${fontSize};
  text-align: ${props => !!props.center ? 'center' : 'initial'};
  margin: 0;
  margin-top: 0.85714em;
  margin-bottom: 0.57142em;
  color: ${palette({ secondary: 3 }, 1)};
`

const Heading = styled(({
  level, children, reverse, palette, theme, ...props
}) => React.createElement(`h${level}`, props, children))`${styles}`

Heading.propTypes = {
  level: T.number,
  children: T.node,
  palette: T.string,
  reverse: T.bool,
  bold: T.string,
  center: T.string,
}

Heading.defaultProps = {
  level: 1,
  palette: 'grayscale',
}

export default Heading