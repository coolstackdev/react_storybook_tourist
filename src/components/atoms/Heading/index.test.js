import 'jest-styled-components'
import styled from 'styled-components'
import Heading from '.'

const wrap = (props = {}) => shallow(<Heading {...props} />).dive()
describe('Button', () => {
  it('should render correctly with no props', () => {
    const component = shallow(<Heading />)
    expect(component).toMatchSnapshot()
  });
})


// it('renders children when passed in', () => {
//   const wrapper = wrap({ children: 'test' })
//   console.warn("wrapper", wrapper)
//   expect(wrapper.contains('test')).toBe(true)
// })

// it('renders props when passed in', () => {
//   const wrapper = wrap({ id: 'foo' })
//   expect(wrapper.find({ id: 'foo' })).toHaveLength(1)
// })

// it('renders h1 by default', () => {
//   const wrapper = wrap()
//   expect(wrapper.find('h1')).toHaveLength(1)
// })

// it('renders hLevel when level is passed in', () => {
//   const wrapper = wrap({ level: 2 })
//   expect(wrapper.find('h2')).toHaveLength(1)
// })
