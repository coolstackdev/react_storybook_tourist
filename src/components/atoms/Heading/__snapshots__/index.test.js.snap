// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly with no props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "_foldedDefaultProps": Object {
        "level": 1,
        "palette": "grayscale",
      },
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Heading-sc-214rfr-0",
        "isStatic": false,
        "rules": Array [
          "
  font-family: ",
          [Function],
          ";
  font-weight: ",
          [Function],
          ";
  font-size: ",
          [Function],
          ";
  margin: 0;
  margin-top: 0.85714em;
  margin-bottom: 0.57142em;
  color: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "Heading",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "bold": [Function],
        "children": [Function],
        "level": [Function],
        "palette": [Function],
        "reverse": [Function],
      },
      "render": [Function],
      "styledComponentId": "Heading-sc-214rfr-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  level={1}
  palette="grayscale"
/>
`;
